addon/session-stores/cookie.js

```diff
@@ -242,8 +242,10 @@ export default BaseStore.extend({
   },
 
   rewriteCookie() {
-    const data = this._read(this._oldCookieName);
-    const expiration = this._calculateExpirationTime();
-    this._write(data, expiration);
+    run.scheduleOnce('actions', this, function() {
+      const data = this._read(this._oldCookieName);
+      const expiration = this._calculateExpirationTime();
+      this._write(data, expiration);
+    });
```

Comment:
The above `run.scheduleOnce('actions', this, this.rewriteCookie);` should be sufficient. I think it's easier if we keep the actual method a synchronous operation.